docker-compose.yml

```diff
@@ -1,82 +1,81 @@
-version: '2'
+version: '3'
 
 volumes:
-    prometheus_data: {}
-    grafana_data: {}
+  prometheus-config:
+    external:
+      name: prometheus-config
+  prometheus-data:
+  grafana-data:
+  alertmanager-config:
+    external:
+      name: alertmanager-config
 
 networks:
   front-tier:
-    driver: bridge
   back-tier:
-    driver: bridge
 
 services:
   prometheus:
     image: prom/prometheus
     container_name: prometheus
     volumes:
-      - ./prometheus/:/etc/prometheus/
-      - prometheus_data:/prometheus
+      - prometheus-config:/etc/prometheus/
+      - prometheus-data:/prometheus
     command:
       - '-config.file=/etc/prometheus/prometheus.yml'
       - '-storage.local.path=/prometheus'
       - '-alertmanager.url=http://alertmanager:9093'
-    expose:
-      - 9090
     ports:
       - 9090:9090
-    links:
-      - cadvisor:cadvisor
-      - alertmanager:alertmanager
     depends_on:
       - cadvisor
     networks:
       - back-tier
-
+    deploy:
+      placement:
+        constraints:
+          - node.role == manager
+          - node.hostname == node1
```

Comment:
Temporarily it's rely on where the volume is created.

Later can remove it, if we find the "overlay volume" container like overly network.